tests/upgrade_test.go

```diff
@@ -0,0 +1,124 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Jan Christoph Uhde <jan@uhdejc.com>
+//
+package tests
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/dchest/uniuri"
+
+	//driver "github.com/arangodb/go-driver"
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	kubeArangoClient "github.com/arangodb/kube-arangodb/pkg/client"
+	"github.com/arangodb/kube-arangodb/pkg/util"
+)
+
+// test upgrade single server mmfiles 3.2 -> 3.3
+func TestUpgradeSingleMMFiles32to33(t *testing.T) {
+	upgradeSubTest(t, api.DeploymentModeSingle, api.StorageEngineMMFiles, "3.2.12", "3.3.3")
+}
+
+// test upgrade single server rocksdb 3.3 -> 3.4
+func TestUpgradeSingleRocksDB33to34(t *testing.T) {
+	upgradeSubTest(t, api.DeploymentModeSingle, api.StorageEngineRocksDB, "3.3.3", "3.4.0")
+}
+
+// test upgrade resilient single server rocksdb 3.2 -> 3.3
+func TestUpgradeResilientSingleRocksDB32to33(t *testing.T) {
+	upgradeSubTest(t, api.DeploymentModeResilientSingle, api.StorageEngineRocksDB, "3.2.12", "3.3.3")
+}
+
+// test upgrade resilient single server mmfiles 3.3 -> 3.4
+func TestUpgradeResilientSingleMMFiles33to34(t *testing.T) {
+	upgradeSubTest(t, api.DeploymentModeResilientSingle, api.StorageEngineMMFiles, "3.3.0", "3.4.0")
+}
+
+// test upgrade cluster rocksdb 3.2 -> 3.3
+func TestUpgradeClusterRocksDB32to33(t *testing.T) {
+	upgradeSubTest(t, api.DeploymentModeCluster, api.StorageEngineRocksDB, "3.2.12", "3.3.3")
+}
+
+// test upgrade cluster mmfiles 3.3 -> 3.4
+func TestUpgradeClusterMMFiles33to34(t *testing.T) {
+	upgradeSubTest(t, api.DeploymentModeCluster, api.StorageEngineRocksDB, "3.3.12", "3.4.0")
+}
+
+func upgradeSubTest(t *testing.T, mode api.DeploymentMode, engine api.StorageEngine, fromVersion, toVersion string) error {
+	// check environment
+	longOrSkip(t)
+
+	k8sNameSpace := getNamespace(t)
+	k8sClient := mustNewKubeClient(t)
+	deploymentClient := kubeArangoClient.MustNewInCluster()
+
+	// Prepare deployment config
+	deploymentName := "test-upgrade-" + string(mode) + "-" + string(engine)
```

Comment:
This looks a bit complicated.

`deploymentName := strings.Replace(fmt.Sprintf("test-upgrade-%s-%s-%sto%s", mode, engine, fromVersion, toVersion), ".", "-", -1)`